scripts/seed_users.js

```diff
@@ -0,0 +1,414 @@
+
+// Set the node enviornment to test
+process.env.NODE_ENV = 'test';
+
+// Define user with parameterized constructor
+class User {
+    constructor(name, type, language, color, password, classroom) {
+        this.name = name;
+        this.type = type;
+        this.language = language;
+        this.color = color;
+        this.password = password;
+        if (classroom)
+            this.classroom = classroom;
+        this.status = 0;
+        this.comment = "";
+        this._id = "";
+    }
+}
+
+// Initialize the array that will contains all the users read from CSV
+var Users = [];
+
+// Trim and lowercase string
+function cleanString(string) {
+    if (!string) return;
+    string = string.trim();
+    string = string.toLowerCase();
+    return string;
+}
+
+// Validate color string
+function isValidColor(color) {
+    try {
+        color = JSON.parse(color);
+    } catch (e) {
+        return false;
+    }
+    if (!color.stroke || !color.fill) return false;
+    return true
+}
+
+// Generate random password
+function generatePassword(length) {
+    var password = "";
+    var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    
+    for (var i = 0; i < length; i++)
+        password += possible.charAt(Math.floor(Math.random() * possible.length));
+    
+    return password;
+}
+
+// Validate data row
+function validateUserRow(user, index) {
+    if (!user.name) {
+        console.log("Row: " + (index + 1) + " -> Username is invalid\nSkipping...");
+        return;
+    }
+
+    user.type = cleanString(user.type);
+    if (!user.type) {
+        user.type = "student";
+    }
+
+    // ToDo: Validate language from an array of valid languages
+    user.language = cleanString(user.language);
+    if (!user.language) {
+        user.language = "en";
+    }
+
+    user.color = cleanString(user.color);
+    if (!isValidColor(user.color)) {
+        // ToDo: Randomize Color
+        user.color = '{ "stroke":"#ff0000", "fill":"#0000ff" }';
+    }
+
+    // ToDo: Get minimum password length from .ini file
+    if (!user.password || typeof user.password != "string" || user.password.length < 4) {
+        console.log("Row: " + (index + 1) + " -> User: " + user.name + " -- Password is invalid\nGenerating random password...");
+        user.password = generatePassword(8);
+    }
+
+    user.classroom = user.classroom || "";
+    user.classroom = user.classroom.trim();
+
+    return user;
+}
+
+function stringifyUser(user) {
+    return '{"name":"' + user.name + '", ' + 
+            '"color":' + user.color + ', ' +
+            '"role":"' + user.type + '", ' +
+            '"password":"' + user.password + '", ' + 
+            '"language":"' + user.language + '", ' +
+            '"options":{"sync":true, "stats":true}}';
+}
+
+// Initiate Master Admin
+var masterAdmin = stringifyUser(new User('Master_Admin_' + (+new Date()).toString(), 'admin', 'en', '{"stroke":"#ff0000", "fill":"#0000ff"}', 'password'));
```

Comment:
Does it mean that the script will create a master admin each time it's run?

Reply:
Yes. The master admin will be deleted when the script finishes.
There's no way to access other API routes without an admin role. I prefer not to modify the existing API.